src/env.cc

```diff
@@ -512,11 +512,9 @@ void TrackingTraceStateObserver::UpdateTraceCategoryState() {
 
 void Environment::AssignToContext(Local<v8::Context> context,
                                   const ContextInfo& info) {
+  ContextEmbedderTag::TagNodeContext(context);
```

Comment:
Technically this should be done at the end of this method and the tagging should not be done before other fields are initialized,  in case the `ContextCreated` call below invokes anything that attempts to access context fields, though it wouldn't be a problem until https://github.com/nodejs/node/pull/44252 which makes it possible to fire `ContextCreated()` on contexts that have a global object with interceptors that attempt to access context fields, and the problem predates this PR anyway , so I am good with leaving it for that PR instead.